### PR TITLE
Update aichat_controller tests / remove logging to AichatSessions

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR removes logging to AichatSessions from the `aichat` controller action `chat_completion`. It also updates the unit tests for `chat_completion` and adds tests for the new `log_chat_event` controller action. There is no impact on the user UI or experience.

Note that this PR will NOT be merged until https://github.com/code-dot-org/code-dot-org/pull/60110 is merged which adds logging to the new 'AichatEvents' table.

## Links
[jira](https://codedotorg.atlassian.net/browse/LABS-952)

## Testing story
Ran updated unit tests locally and they passed. Tested a couple aichat levels in /allthethings to confirm no changes to user.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/ability.rb
Comment (fisher-alice) [inline dashboard/app/models/ability.rb:484]: Note that this typo update is included here so that unit tests could pass as expected for this PR.
This typo fix is also included in https://github.com/code-dot-org/code-dot-org/pull/60110.

**Review by sanchitmalhotra126 — APPROVED** [review]
🎉 for code cleanup!

Do we want to hold off on merging this for a bit even after the front end changes are complete, so we can verify that data is being logged correctly before removing logging to the previous table?

**Comment by fisher-alice** [comment]
> 🎉 for code cleanup!
> 
> Do we want to hold off on merging this for a bit even after the front end changes are complete, so we can verify that data is being logged correctly before removing logging to the previous table?

yes - definitely! Will confirm logging to new table first before merging this PR.

**Comment by fisher-alice** [comment]
> > 🎉 for code cleanup!
> > Do we want to hold off on merging this for a bit even after the front end changes are complete, so we can verify that data is being logged correctly before removing logging to the previous table?
> 
> yes - definitely! Will confirm logging to new table first before merging this PR.

@sanchitmalhotra126  - Confirmed that logging in AichatEvents table after https://github.com/code-dot-org/code-dot-org/pull/60110 was merged yesterday. 🎉 

Screenshot 2024-08-09 at 7 56 50 AM

**Pull request merged**
